### ISSUE TITLE
switch case indentation formatting broken in ifdefs

### ISSUE THREAD

**Issue by mdznr** [nicklockwood/SwiftFormat#847]
No matter what formatting options for `indent` I select, the formatting for case statements _inside_ of `ifdef`s is broken.

**Input (and expected output):**
```swift
public struct Testing {
	enum Example {
		case foo
		case bar
	}
	
	private func baz(value: Example) -> String {
		#if DEBUG
		switch value {
			case .foo: return "foo"
			case .bar: return "bar"
			@unknown default: return "unknown"
		}
		#else
		switch value {
			case .foo: return "foo"
			case .bar: return "bar"
			@unknown default: return "unknown"
		}
		#endif
		switch value {
			case .foo: return "foo"
			case .bar: return "bar"
			@unknown default: return "unknown"
		}
	}
}
```

**.swiftformat:**
```
--ifdef indent
--indent tab
--indentcase true
--smarttabs enabled
--tabwidth unspecified
--xcodeindentation disabled
```

**Output:**
```swift
public struct Testing {
	enum Example {
		case foo
		case bar
	}
	
	private func baz(value: Example) -> String {
		#if DEBUG
			switch value {
			case .foo: return "foo"
			case .bar: return "bar"
				@unknown default: return "unknown"
			}
		#else
			switch value {
			case .foo: return "foo"
			case .bar: return "bar"
				@unknown default: return "unknown"
			}
		#endif
		switch value {
			case .foo: return "foo"
			case .bar: return "bar"
			@unknown default: return "unknown"
		}
	}
}
```

The `switch` and `case` statements _inside_ the `ifdef`s are inconsistent with same statements not in an `ifdef`.

Version 0.47.10

**Comment by nicklockwood** [comment]
(reverted fix as it broke other things - will try again later)

**Comment by PompeiaPaetenari** [comment]
We have this issue as well and disable swiftformat for the few lines it happens with.  As they are the last few `swiftformat:disable` lines in our codebase, I would love to remove them, too.

tl;dr: adding a comment to get notified on resolution.

**Comment by nicklockwood** [comment]
@mdznr @PompeiaPaetenari fixed in 0.47.12
